modin/backends/pandas/query_compiler.py

```diff
@@ -1559,6 +1559,19 @@ def describe(self, **kwargs):
             .astype(self.dtypes)
             .describe(**kwargs)
         )
+        new_index = empty_df.index
+
+        # Note: `describe` convert timestamp type to object type
+        # which results in the loss of two values in index: `first` and `last`
+        # for empty DataFrame.
+        datetime_is_numeric = kwargs.get("datetime_is_numeric") or False
+        if not any(map(is_numeric_dtype, empty_df.dtypes)) and not datetime_is_numeric:
+            for col_name in empty_df.dtypes.index:
+                # if previosly type of `col_name` was datetime or timedelta
+                if is_datetime_or_timedelta_dtype(self.dtypes[col_name]):
+                    new_index = pandas.Index(
+                        empty_df.index.to_list() + ["first"] + ["last"]
+                    )
```

Comment:
shouldn't we add `break` here to don't do extra iterations when we're already found out that we need two extra indices?

Reply:
Good catch! Should